torch_geometric/nn/summary.py

```diff
@@ -0,0 +1,156 @@
+from collections import defaultdict
+from time import perf_counter
+from typing import Any, Sequence
+
+import torch
+import torch.nn as nn
+
+
+def summary(
+    model: nn.Module,
+    *inputs: Sequence[Any],
+    device: torch.device = 'cpu',
+    max_depth: int = 3,
```

Comment:
In addition to `max_depth`, we could think about adding a function about which layers should be treated as "leaf" modules. For example, it probably makes sense to let any instance of `MessagePassing` be treated as a leaf module. WDYT?

Reply:
Not sure I understand you, but I'll have a try.